src/main/java/store/MemoryProductStorage.java

```diff
@@ -0,0 +1,64 @@
+package store;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class MemoryProductStorage implements ProductStorage {
+
+	List<Produit> productList = new ArrayList<>();
+
+	@Override
+	public String save(Produit p) {
+		p.id = UUID.randomUUID().toString();
+		productList.add(p);
+		return p.id;
+	}
+
+	@Override
+	public void update(String id, Produit p) {
+		Produit product = getByID(id);
+		int productIndex = productList.indexOf(product);
+		p.id = UUID.randomUUID().toString();
+		productList.add(productIndex, p);
```

Comment:
Could the new product be added at the end of the list, this saving the computation brought in by  `.indexOf`?

Reply:
Non... on passe par une map et on trouve direct le bon objet. pas besoin de mettre à la fin